packages/api/src/modules/external-integrations/guards/external-integrations-auth.guard.ts

```diff
@@ -0,0 +1,31 @@
+import {
+  BadRequestException,
+  CanActivate,
+  ExecutionContext,
+  Injectable,
+  UnauthorizedException,
+} from '@nestjs/common';
+
+import { UsersService } from '../../users/users.service';
+
+@Injectable()
+export class ExternalIntegrationsAuthGuard implements CanActivate {
+  constructor(private readonly usersService: UsersService) {}
+
+  public async canActivate(context: ExecutionContext): Promise<boolean> {
+    const request = context.switchToHttp().getRequest<{ headers: ExternalIntegrationsHeaders }>();
+    const { authorization, from } = request.headers;
+
+    if (!authorization || !from) {
+      throw new BadRequestException();
+    }
+
+    const foundUser = await this.usersService.findOneByEmail(from);
+
+    if (!foundUser || authorization !== foundUser.externalIntegrationsToken) {
```

Comment:
when you remove a token you set it to '' (empty string). What if  someone sends you an email with an empty token??? XD